src/librustdoc/html/format.rs

```diff
@@ -1064,7 +1074,11 @@ impl clean::Impl {
                 write!(f, " for ")?;
             }
 
-            if let Some(ty) = self.kind.as_blanket_ty() {
+            if let clean::Type::Tuple(types) = &self.for_ &&
+                let [clean::Type::Generic(name)] = &types[..] &&
+                (self.kind.is_tuple_varadic() || self.kind.is_auto()) {
+                primitive_link_fragment(f, PrimitiveType::Tuple, &format!("({name}, ...)"), "#trait-implementations-1", cx)?;
```

Comment:
Just a note to say that we are likely to eventually scope anchor fragments, such that this anchor would change, and the internal link would break. Here's a possible way to catch that: Write a test case that contains a `# Trait Implementations` heading, and verify that it contains this anchor. The test case could then have a comment indicating what needs to be updated.